master/internal/api_experiment.go

```diff
@@ -852,36 +904,106 @@ func (a *apiServer) PauseExperiment(
 	}
 }
 
+func (a *apiServer) PauseExperiments(
+	ctx context.Context, req *apiv1.PauseExperimentsRequest,
+) (*apiv1.PauseExperimentsResponse, error) {
+	query, err := a.editableExperimentIds(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	expIDs := []int32{}
+	err = query.Where("id IN (?)", bun.In(req.ExperimentIds)).
+		Model(&expIDs).
+		Scan(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &apiv1.PauseExperimentsResponse{ExperimentIds: []int32{}}
+	mockResp := &apiv1.PauseExperimentResponse{}
+	for _, expID := range expIDs {
+		addr := experimentsAddr.Child(expID)
+		err = a.ask(addr, &apiv1.PauseExperimentRequest{Id: expID}, mockResp)
+		if err == nil {
+			resp.ExperimentIds = append(resp.ExperimentIds, expID)
+		}
+	}
+
+	return resp, nil
+}
+
 func (a *apiServer) CancelExperiment(
 	ctx context.Context, req *apiv1.CancelExperimentRequest,
 ) (resp *apiv1.CancelExperimentResponse, err error) {
-	if _, _, err = a.getExperimentAndCheckCanDoActions(ctx, int(req.Id),
-		expauth.AuthZProvider.Get().CanEditExperiment); err != nil {
+	_, err = a.CancelExperiments(ctx,
+		&apiv1.CancelExperimentsRequest{ExperimentIds: []int32{req.Id}})
+	return &apiv1.CancelExperimentResponse{}, err
```

Comment:
I assume in the past if there was an error updating the single object, the API would return an error.
Bulk API just does the operation on a subset of objects it can update.
does it make sense to check here if the single requested object has been updated successfully?

Reply:
Was thinking about:
- creating an error here in place of the missing one (a little frustrating because we can report but won't know the actual issue)
- having CancelExperiments return the specific error `if len(req.ExperimentIds) == 1`

Reply:
Shouldn't we be returning the errors from bulk APIs like `CancelExperiments` as well? 

Imagine, as an API user, you're sending a request to cancel 10 experiments, response has only one exp.id, the other 9 didn't get cancelled. Informing the user why that happened should be valuable.